src/suites/cts/validation/createBindGroupLayout.spec.ts

```diff
@@ -111,17 +107,121 @@ g.test('dynamic set to true is allowed only for buffers', async t => {
   const success = bindingTypeInfo[type].type === 'buffer';
 
   const descriptor = {
-    bindings: [
-      {
-        binding: 0,
-        visibility: GPUShaderStage.FRAGMENT,
-        type,
-        hasDynamicOffset: true,
-      },
-    ],
+    bindings: [{ binding: 0, visibility: GPUShaderStage.FRAGMENT, type, hasDynamicOffset: true }],
   };
 
   t.expectValidationError(() => {
     t.device.createBindGroupLayout(descriptor);
   }, !success);
 }).params(poptions('type', bindingTypes));
+
+{
+  function pickExtraBindingTypes(type: GPUBindingType, extraTypeSame: boolean): GPUBindingType[] {
+    if (extraTypeSame) {
+      switch (type) {
+        case 'storage-buffer':
+        case 'readonly-storage-buffer':
+          return ['storage-buffer', 'readonly-storage-buffer'];
+        default:
+          return [type];
+      }
+    } else {
+      return type === 'sampler' ? ['sampled-texture'] : ['sampler'];
+    }
+  }
+
+  const kCasesForMaxResourcesPerStageTests: ParamSpec[] = [];
```

Comment:
nit: This could be declared outside of the { } scope, as well as the tests. This helps the tests be indented the same, and hides `pickExtraBindingTypes` when it's strictly used to populate `kCasesForMaxResourcesPerStageTests`